packages/firestore/test/unit/api/database.test.ts

```diff
@@ -80,32 +80,39 @@ describe('Query', () => {
 describe('QuerySnapshot', () => {
   it('support equality checking with isEqual()', () => {
     expectEqual(
-      querySnapshot('foo', {}, { a: { a: 1 } }, true, false, false),
-      querySnapshot('foo', {}, { a: { a: 1 } }, true, false, false)
+      querySnapshot('foo', {}, { a: { a: 1 } }, keys('coll/a'), false, false),
```

Comment:
"coll/a" seems to be unrelated to the collection in question.

Also, it's weird that this can have mutated keys but neither from cache nor sync state changed flags set.

Reply:
I changed these tests to use `foo`.

I also went through and removed the mutations from most of these comparisons. Only the one case that directly compared the mutated keys still has mutated keys with both flags false (to make it obvious what is tested here). 
BTW, I think this case could happen, albeit it would be rare (the backend has sent us all committed documents, but we are still not CURRENT since there are more edits from other clients).

I also fixed the ViewSnapshot comparison to take `mutatedKeys` into account.